README.md

```diff
@@ -1,28 +1,149 @@
+settings.option_defer = Some(0);
+settings.set("option_defer", b"13")?;
+settings.set("option_miniconf/a", b"14").unwrap_err();
+*settings.option_miniconf = Some(Inner::default());
+settings.set("option_miniconf/a", b"14")?;
+
+// Serializing an element by path
+let len = settings.get("foo", &mut buf)?;
+assert_eq!(&buf[..len], b"true");
+
+// Iterating over all elements
+for path in Settings::iter_paths::<3, 32>().unwrap() {
+    let len = settings.get(&path, &mut buf)?;
+    if path.as_str() == "option_miniconf/a" {
+        assert_eq!(&buf[..len], b"14");
+    }
+}
+
+# Ok::<(), miniconf::Error>(())
+```
+
+## MQTT
+There is an [MQTT-based client](MqttClient) that implements settings management over the [MQTT
+protocol](https://mqtt.org) with JSON payloads. A Python reference library is provided that
+interfaces with it.
 
 ```sh
-# Set the `sample_rate_hz` value of device with identifier `quartiq/example_device` to `10`.
-python -m miniconf quartiq/example_device sample_rate_hz=10
+# Discover the complete unique prefix of an application listening to messages
+# under the topic `quartiq/application/12345` and set its `foo` setting to `true`.
+python -m miniconf -d quartiq/application/+ foo=true
 ```
+
+## Design
+For structs with named fields, Miniconf offers a [derive macro](derive.Miniconf.html) to automatically
+assign a unique path to each item in the namespace of the struct.
+The macro implements the [Miniconf] trait that exposes access to serialized field values through their path.
+All types supported by [serde_json_core] can be used as fields.
+
+Elements of homogeneous [core::array]s are similarly accessed through their numeric indices.
+Structs, arrays, and Options can then be cascaded to construct a multi-level namespace.
+Namespace depth and access to individual elements instead of the atomic updates
+is configured at compile (derive) time using the `#[miniconf(defer)]` attribute.
+`Option` is used with `#[miniconf(defer)]` to support paths that may be absent (masked) at
+runtime.
+
+While the [Miniconf] implementations for [core::array] and [core::option::Option] by provide
+atomic access to their respective inner element(s), [Array] and
+[Option] have alternative [Miniconf] implementations that expose deep access
+into the inner element(s) through their respective inner [Miniconf] implementations.
+
+## Formats
+The path hierarchy separator is the slash `/`.
+
+Values are serialized into and deserialized from JSON.
+
+## Transport
+Miniconf is designed to be protocol-agnostic. Any means that can receive key-value input from
+some external source can be used to modify values by path.
+
+## Limitations
+Deferred (non-atomic) access to inner elements of some types is not yet supported. This includes:
+* Complex enums (other than [core::option::Option])
+* Tuple structs (other than [Option], [Array])
+
+## Features
+* `mqtt-client` Enabled the MQTT client feature. See the example in [MqttClient].
```

Comment:
While these links are valid for `rust-doc`, they are not valid markdown. As such, most of the links to rust code in this doc are broken:

![image](https://user-images.githubusercontent.com/8771450/200007363-212576ae-70ce-4d81-b2f2-c2c42fd6911a.png)

Reply:
I know. But I think that's a minor issue, less critical than having redundant and divergent docs in README vs the library docstrings. I'm happy to have a slightly ugly README if in turn the content is much better and in sync with the docs.

Reply:
Fair enough. We can address it later if it becomes annoying